File: LeetCode/0128.Longest-Consecutive-Sequence/Longest-Consecutive-Sequence.java
class Solution {
    public int longestConsecutive(int[] nums) {
        Set<Integer> set = new HashSet<>();
        for(int i: nums){
            set.add(i);
        }
        int res = 0;
        for(int i = 0; i < nums.length; i++){
            if(!set.contains(nums[i])) continue;
            set.remove(nums[i]);
            
            int lower = nums[i] - 1;
            //ex: 4 -> 3
            while(set.contains(lower)){
            //find!!
            set.remove(lower);
            lower = lower - 1;
            }

            int upper = nums[i] + 1;
            //4 of upper is 5
            while(set.contains(upper)){
                //find!!!
                set.remove(upper);
                upper ++;
            }
            //lower = 3, 2, 1 -> 0(invalid)
            //upper = 5(invalid)
            //4, 3, 2, 1 
            res = Math.max(res, upper - lower - 1);
        }
        return res;
    }
}

File: LeetCode/0128.Longest-Consecutive-Sequence/Longest-Consecutive-Sequence.py
class Solution(object):
    def longestConsecutive(self, nums):
        """
        :type nums: List[int]
        :rtype: int
        """
        nset = set(nums)
        res = 0
        for num in nums:
            if num not in nset: continue
            nset.remove(num)
            left, right = num - 1, num + 1
            while left in nset:
                nset.remove(left)
                left -= 1
            while right in nset:
                nset.remove(right)
                right += 1
            res = max(res, right - left - 1)
        return res
